src/lib/converter/input/witnessUtxo.js

```diff
@@ -61,20 +84,30 @@ function encode(data) {
   varuint.encode(script.length, result, resultLen);
   resultLen += varintLen;
   script.copy(result, resultLen);
+  if (rangeProof && surjectionProof) {
+    varuint.encode(rangeProof.length, result, resultLen);
+    resultLen += varuint.encodingLength(rangeProofLen);
+    rangeProof.copy(result, resultLen);
+    varuint.encode(surjectionProof.length, result, resultLen);
+    resultLen += varuint.encodingLength(surjectionProofLen);
+    surjectionProof.copy(result, resultLen);
+  }
   return {
     key: Buffer.from([typeFields_1.InputTypes.WITNESS_UTXO]),
     value: result,
   };
 }
 exports.encode = encode;
 exports.expected =
-  '{ script: Buffer; value: Buffer; asset: Buffer; nonce: Buffer; }';
+  '{ script: Buffer; value: Buffer; asset: Buffer; nonce: Buffer; rangeProof?: Buffer; surjectionProof?: Buffer }';
 function check(data) {
   return (
     Buffer.isBuffer(data.script) &&
     Buffer.isBuffer(data.value) &&
     Buffer.isBuffer(data.asset) &&
-    Buffer.isBuffer(data.nonce)
+    Buffer.isBuffer(data.nonce) &&
+    (data.rangeProof ? Buffer.isBuffer(data.rangeProof) : true) &&
```

Comment:
Why we return true?

Reply:
is like: if there is no rangeproof then skip (return `true`)

Reply:
Add a comment to explain the "hack" for sake of brevity